pkg/kubelet/status/generate_test.go

```diff
@@ -115,6 +115,103 @@ func TestGeneratePodReadyCondition(t *testing.T) {
 	}
 }
 
+func TestGeneratePodInitializedCondition(t *testing.T) {
+	tests := []struct {
+		spec              *v1.PodSpec
+		containerStatuses []v1.ContainerStatus
+		podPhase          v1.PodPhase
+		expected          v1.PodCondition
+	}{
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: nil,
+			podPhase:          v1.PodRunning,
+			expected:          getInitializedCondition(false, "UnknownContainerStatuses", ""),
+		},
+		{
+			spec:              &v1.PodSpec{},
+			containerStatuses: []v1.ContainerStatus{},
+			podPhase:          v1.PodRunning,
+			expected:          getInitializedCondition(true, "", ""),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{},
+			podPhase:          v1.PodRunning,
+			expected:          getInitializedCondition(false, "ContainersNotInitialized", "containers with unknown status: [1234]"),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getReadyStatus("1234"),
+				getReadyStatus("5678"),
+			},
+			podPhase: v1.PodRunning,
+			expected: getInitializedCondition(true, "", ""),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getReadyStatus("1234"),
+			},
+			podPhase: v1.PodRunning,
+			expected: getInitializedCondition(false, "ContainersNotInitialized", "containers with unknown status: [5678]"),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+					{Name: "5678"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getReadyStatus("1234"),
+				getNotReadyStatus("5678"),
+			},
+			podPhase: v1.PodRunning,
+			expected: getInitializedCondition(false, "ContainersNotInitialized", "containers with incomplete status: [5678]"),
+		},
+		{
+			spec: &v1.PodSpec{
+				InitContainers: []v1.Container{
+					{Name: "1234"},
+				},
+			},
+			containerStatuses: []v1.ContainerStatus{
+				getReadyStatus("1234"),
+			},
+			podPhase: v1.PodSucceeded,
+			expected: getInitializedCondition(true, "PodCompleted", ""),
+		},
+	}
+
+	for i, test := range tests {
+		condition := GeneratePodInitializedCondition(test.spec, test.containerStatuses, test.podPhase)
+		if !reflect.DeepEqual(condition, test.expected) {
```

Comment:
To clarify - it's the `Message` filed that's the problem, not the `Reason` field. The problem with the message field is that it is a human-readable string with a poorly defined format. It means that if someone decides to log something slightly different then they need to update it in two places (one being this test).

Rather than using a DeepEqual, I suggest testing the individual fields (there's only 3 that need to be tested).